server/index.js

```diff
@@ -111,78 +114,38 @@ app.post("/volume", isIdentified, (req, res) => {
 
 app.post("/skip", (req, res) => {
   const { body } = req;
-  if (typeof body.id === "number") {
-    queue = queue
-      .map(
-        e =>
-          body.id === e.id
-            ? { ...e }
-            : { ...e, votes: e.votes.concat([body.token]) }
-      )
-      .filter(e => shouldSkip(e.votes.length, users.length));
+  if (typeof body.id === "string") {
+    const item_voted = queue.find(e => body.id === e.id)
+    
+    if (item_voted) {
+        item_voted.votes.add(req.token)
+        queue = queue.filter(e => shouldSkip(e.votes.size, users.size));
+        res.status(204).end();
+    }
+    else {
+      res.status(404).json({ error: "Id não encontrado" });
+    }
   } else {
-    res.status(400).json({ error: "O id do voto deve ser um número!" });
+    res.status(400).json({ error: "O id do voto deve ser uma string!" });
   }
 });
 
-app.get("/identify", (req, red) => {
-  res.json({ users: users.length });
-});
-
-app.post("/identify", (req, res) => {
-  const { body } = req;
-  if (typeof body.token === "undefined") {
-    const token = uuid();
-    users = users.concat({ token, timer: 1 * 60 });
-    res.status(200).json({ token });
-  } else if (typeof body.token === "string") {
-    if (!users.find(u => u.token === body.token)) {
-      res.status(401).json({ error: "O token é inválido!" });
-    } else {
-      users = users.map(u => {
-        if (u.token === body.token) u.timer = 1 * 60;
-        return u;
-      });
-      res.status(200).json({ token: body.token });
-    }
-  } else {
-    res.status(400).json({
-      error:
-        "O token deve ser uma string ou deve ser vazio para receber um novo!"
-    });
-  }
+app.get("/identify", (req, res) => {
+  res.json({ users: users.size });
 });
 
 setInterval(() => {
-  users = users
-    .map(u => {
-      u.timer -= 1;
-      return u;
-    })
-    .filter(u => u.timer > 0);
+  users.forEach((time, token) => {
+    const new_time = time - 1;
+    if (new_time) users.set(token, new_time);
```

Comment:
Why'd you update a map while iterating through it?

Reply:
Why not?